src/bytes/bytes_lemmas.gobra

```diff
@@ -0,0 +1,359 @@
+ensures  forall j int :: { j in indices }{ s[j:j+len(sep)] } !(j in indices) ==> ( !InRangeInc(j, 0, len(s) - len(sep)) || s[j:j+len(sep)] != sep || SetContainsInRange(indices, j-len(sep), j))
+decreases
+func lemmaCountAux(s, sep seq[byte], indices set[int], idx int) {
+	assume forall j int :: {s[j:j+len(sep)]} InRangeInc(j, idx, len(s) - len(sep)) ==> s[j:j+len(sep)] != sep
+	assert forall j int :: { j in indices }{ s[j:j+len(sep)] } !(j in indices) && InRangeInc(j, 0, idx - len(sep)) && !SetContainsInRange(indices, j-len(sep), j) ==> s[j:j+len(sep)] != sep
+	assert forall j int :: { j in indices }{ s[j:j+len(sep)] } !(j in indices) && InRangeInc(j, idx, len(s) - len(sep)) && !SetContainsInRange(indices, j-len(sep), j) ==> s[j:j+len(sep)] != sep
+	assume forall j int :: { j in indices }{ s[j:j+len(sep)] } !(j in indices) && InRangeInc(j, 0, len(s) - len(sep)) && !SetContainsInRange(indices, j-len(sep), j) ==> s[j:j+len(sep)] != sep
+}
+
+
+
+// Repeat
+
+ghost
+ensures bytes.Repeat(b, 1) == b
+decreases
+func lemmaRepeatOnce(b seq[byte]) {
+	assert bytes.Repeat(b, 1) == b ++ bytes.Repeat(b, 0)
+}
+
+ghost
+requires n > 0
+ensures bytes.Repeat(b, n) == bytes.Repeat(b, n-1) ++ b
+decreases n
+func lemmaRepeatFlip(b seq[byte], n int) {
+	if n == 1 {
+		assert bytes.Repeat(b, n-1) == seq[byte]{}
+	} else {
+		assert bytes.Repeat(b, n) == b ++ bytes.Repeat(b, n-1)
+		lemmaRepeatFlip(b, n-1)
+		assert bytes.Repeat(b, n-1) == bytes.Repeat(b, n-2) ++ b
+	}
+}
+
+ghost
+requires n >= 0
+ensures  b ++ bytes.Repeat(b, n) == bytes.Repeat(b, n) ++ b
+decreases
+func lemmaRepeatCommutative(b seq[byte], n int) {
+	assert b ++ bytes.Repeat(b, n) == bytes.Repeat(b, n+1)
+	lemmaRepeatFlip(b, n+1)
+}
+
+ghost
+requires n >= 0
+ensures  bytes.Repeat(b, 2 * n) == bytes.Repeat(b, n) ++ bytes.Repeat(b, n)
+decreases n
+func lemmaRepeat_2n(b seq[byte], n int) {
+	if n != 0 {
+		assert bytes.Repeat(b, n) == b ++ bytes.Repeat(b, n - 1)
+		assert bytes.Repeat(b, n) ++ bytes.Repeat(b, n) == b ++ bytes.Repeat(b, n - 1) ++ b ++ bytes.Repeat(b, n - 1)
+		// order of repeating doesn't matter because it's always the same element that is repeated
+		assert b ++ bytes.Repeat(b, n - 1) ++ b ++ bytes.Repeat(b, n - 1) == b ++ (bytes.Repeat(b, n - 1) ++ b) ++ bytes.Repeat(b, n - 1)
+		lemmaRepeatCommutative(b, n - 1)
+		assert b ++ (bytes.Repeat(b, n - 1) ++ b) ++ bytes.Repeat(b, n - 1) == b ++ (b ++ bytes.Repeat(b, n - 1)) ++ bytes.Repeat(b, n - 1)
+		assert b ++ bytes.Repeat(b, n - 1) ++ b ++ bytes.Repeat(b, n - 1) == b ++ b ++ bytes.Repeat(b, n - 1) ++ bytes.Repeat(b, n - 1)
+		lemmaRepeat_2n(b, n - 1)
+		assert bytes.Repeat(b, n - 1) ++ bytes.Repeat(b, n - 1) == bytes.Repeat(b, 2 * (n - 1))
+		assert bytes.Repeat(b, 2 * n - 1) == b ++ bytes.Repeat(b, n - 1) ++ bytes.Repeat(b, n - 1)
+		assert bytes.Repeat(b, 2 * n) == b ++ b ++ bytes.Repeat(b, n - 1) ++ bytes.Repeat(b, n - 1)
+	} else {
+		assert bytes.Repeat(b, 2 * 0) == bytes.Repeat(b, 0) ++ bytes.Repeat(b, 0)
+	}
+}
```

Comment:
I expected that this would be part of the gobra-libs, given that 1) these lemmas describe general properties of `bytes.Repeat` that may be useful to others, 2) they do not mention anything specific to this package.